### DISCUSSION TITLE
How to change joint names in yarpmotorgui

### DISCUSSION THREAD

**Discussion by hu-yue** [robotology/community#76] [Q&A] [answered]
Hi,
How can one change the name of the joints in the yarpmotorgui? I mean the ones in parenthesis after the joint number, like `JOINT 0 (torso_roll)`.
Because in our robot (iCubHeidelberg01) the joints in the gui and the real joints do not match.
In the gui the sequence is `roll - pitch - yaw` but on the real robot it is `yaw - roll - pitch`, and it makes the motorgui quite misleading everytime.
Thanks!

**Comment by pattacini** [comment]
@hu-yue I think we already reported these incongruities in https://github.com/robotology/icub-main/issues/229.

**Comment by hu-yue** [comment]
@pattacini yes thanks!

**Comment by barbalberto** [comment, marked as answer]
`yarpmotorgui` gets the axes names at run-time from `robotInterface`.

The name of the axis is thus specified in the corresponding file `hardware/motorControl/[part_name].xml` by means of the following parameter:

``` xml
<param name="AxisName"> "name_axis_0" "name_axis_1" ... </param>
```

These names will be also used for ROS topics as well.

**Comment by hu-yue** [comment]
@barbalberto Thanks. 
I found the parameter in `mechanicals` instead of `motorControl`.
However this arises a further question about the influence of the parameter. I mean, does this change affects the software of the robot?

**Comment by pattacini** [comment]
Uhm, in `iCubGenova01` they're in `motorControl` ... it seems there are discrepancies then.

**Comment by barbalberto** [comment]
Probably there is some difference between can-based and eth-based robots.
This difference however does not affect the robot behaviour because those two files will be 'merged' together by the code before calling the actual device.

**Comment by pattacini** [comment]
It seems @barbalberto guessed well: `iCubGenova04` (ETH robot) has also those parameters under `mechanicals`.

**Comment by randaz81** [comment]
@yue please do not manually edit those files. They will be fixed soon. (robotology/icub-main#229)

**Comment by hu-yue** [comment]
@randaz81 I will wait for the fix. Thanks!

**Comment by pattacini** [comment]
Closing this issue again then.
